jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -203,8 +227,15 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
             Additional keyword arguments passed to the NDDataArray collapse operation.
             Examples include ``propagate_uncertainties`` and ``operation_ignores_mask``.
         """
+        if self.conflicting_aperture_and_function:
+            self.hub.broadcast(SnackbarMessage(self.conflicting_aperture_error_message,
+                                               color="error", sender=self))
+            raise ValueError(self.conflicting_aperture_error_message)
+            return
```

Comment:
I thought we concluded that snackbar message is unnecessary here or something?

Also, you don't need a return after raise.

Reply:
Moved this to the cone aperture section of `get_aperture` and removed the snackbar message.